### DISCUSSION TITLE
[SUPPORT] Can't play videos

### DISCUSSION THREAD

**Discussion by Frn747** [arvvoid/plugin.video.hbogoeu#207] [Q&A] [answered]
* [X] Are you running the latest version?

**Befor sending the support request this resources can be checked:**

[Kodi Forum thread](https://forum.kodi.tv/showthread.php?tid=339798) |
[Project wiki](https://github.com/arvvoid/plugin.video.hbogoeu/wiki) | 
[FAQ](https://github.com/arvvoid/plugin.video.hbogoeu/wiki/FAQ) | 
[Troubleshoot](https://github.com/arvvoid/plugin.video.hbogoeu/wiki/Troubleshoot)

* [X] I have already checked all resources above

* [ ] I just want to provide feedback or get in touch

**Please enter your question or feedback**

Hi there.

I am quite sure that probably this is not an issue of this addon but I humbly ask for support as I can't make it work to play HBO videos.
I followed the installation steps but when I try to open any video nothing happens, just a quick blink as if it is trying to start but nothing afterwards.
I am running Kody on RPi3. Please find attached the log. 

Any kind soul can help me? Thanks in advance!

[kodi.log](https://github.com/arvvoid/plugin.video.hbogoeu/files/6446383/kodi.log)

**Comment by g1sm** [comment]
Hi,

it seems the issue is related to Widevine CDM library version 13816.64.0. I restored the previous version (13597.105.0) and everything went back to normal.

To restore, go to the HBOEU add-on **_config_** - **_DRM_** - **_inputstream helper settings_** - _**restore widevine CDM library**_ and pick 13597.105.0.

**Comment by Frn747** [comment]
Hi.

Thanks for the reply. I've tried your instructions but I get an error "No backup files exist" (free translation!).
Mine was a new installation so there is no history versions to restore. Is there a way to install the previous version manually?

Thanks.

**Comment by g1sm** [comment]
See [this](https://blog.vpetkov.net/2019/07/12/netflix-and-spotify-on-a-raspberry-pi-4-with-latest-default-chromium/comment-page-3/#comment-459049) comment from Guido on how to download and extract libwidevinecdm.so. The only difference is that you should use ```https://dl.google.com/dl/edgedl/chromeos/recovery/chromeos_13597.105.0_elm_recovery_stable-channel_mp-v2.bin.zip``` as source (replace the url in the first line of the script - the one with wget).

After you have the libwidevinecdm.so file, copy it into the ```/storage/.kodi/cdm/``` folder. Haven't tried this approach, but it should work. Otherwise, just wait a couple of days until the issue is fixed.

**Comment by Frn747** [comment]
Hi G1sm.
Thanks for your support. I think I'll try the lazy solution first and wait a couple of days!

**Comment by arvvoid** [comment]
Hello @Frn747, just tested with the latest version of widevine got with ISH and indeed that seems to be the problem, don't have time to look in detail right now. It seems there are some ongoing changes in the distribution of widevine for some platforms. Read more here, https://github.com/emilsvennesson/script.module.inputstreamhelper/issues/428

Like said above by @g1sm for now the easiest solution is to revert to the previous version of widevine for those that can or use the above described process to obtain the previous version manually.
If not, just tested with the version in the raspbian repository and that work fine: https://archive.raspberrypi.org/debian/pool/main/w/widevine/
if going that route, download and unpack widevine_4.10.1679.0.orig.tar.xz and copy libwidevinecdm.so  into the /storage/.kodi/cdm/ folder on your rpi. Restart your device, playback should work fine.

It looks like a solution for this might take more than a couple of days.

**Comment by arvvoid** [comment]
There is a new release of ISH https://github.com/emilsvennesson/script.module.inputstreamhelper/releases/tag/v0.5.3 so the problem should be solved and no manual intervention should be needed.

**Comment by arvvoid** [comment, marked as answer]
Usually this is due to widevine/inputstream.adaptive issues.

**Comment by arvvoid** [comment]
Old widevine versions have been revoked on ARM Linux platforms for most services, HBO Go is affected.
Unfortunately for ARM devices (Android is not affected) to support new widevine versions an os update is needed or at least some libraries on the OS have to be updated/replaced.
So you have to check on your OS forum/web if an os update supporting new widevine versions is available.

Nothing to do about this on the add-on side.

Take a read here: https://www.matthuisman.nz/2021/05/widevine-arm-broken.html

Extrapolated from matthuisman post:

See below on the status of the various OS's
**LibreELEC 9.2.X (Kodi 18)**
Supported since LibreELEC 9.2.8

**LibreELEC 10.0.X (Kodi 19)**
Supported since LibreELEC 10.0.0

**CoreELEC 9.2.X (Kodi 18)**
Supported since CoreELEC 9.2.8 (September 7th, 2021)

**CoreELEC 19.X (Kodi 19)**
Supported since CoreELEC 19.2-Matrix_rc2 (June 6th, 2021)

**Xbian (Kodi 19)**
Supported since Bullseye update
Update via the XBian-config Kodi add-on.

**Raspberry Pi OS**
Supported since 07/09/2021
Just need to upgrade via terminal
https://www.raspberrypi.org/forums/viewtopic.php?f=63&t=318893&p=1910851#p1910748

**OSMC (Kodi 19)**
Supported since 20/09/2021
Just need to update via OSMC Settings
https://discourse.osmc.tv/t/widevine-add-ons-stopped-working-from-01-09/90684

**OSMC (Kodi 18)**
Won't be officially fixed as no longer supported.
Either update to Kodi 19, or run [THESE](https://github.com/xbmc/inputstream.adaptive/issues/678#issuecomment-913047151) terminal commands

**Other**
You'll need to ask in your OS forums for them to add support.
If your OS is based on Debian, the commands [HERE](https://www.raspberrypi.org/forums/viewtopic.php?f=63&t=318893&p=1910100#p1910100) may help

If after the os update you still experience problems make sure widevine is updated to the latest version via inputstream.helper, you can find a shortcut through the DRM section in the add-on config.

All other OS except ARM linux should be fine if inputstream.adaptive and latest widevine are installed.

At the time of writing most distributions have official or unofficial solutions to the problem.
The situation should be completely resolved when all arm linux distributions support new widevine versions.
This situations can happen and will probably happen again in the future since there is no official widevine support on those devices.
